manager/integration/tests/test_stress.py

```diff
@@ -0,0 +1,316 @@
+        time.sleep(RETRY_INTERVAL)
+    assert pvc_status_is_bound
+
+
+def write_data(k8s_api_client, pod_name):
+    test_data = generate_random_data(TEST_DATA_BYTES)
+    write_pod_volume_data(k8s_api_client, pod_name, test_data)
+    volume_data = read_volume_data(k8s_api_client, pod_name)
+    assert test_data == volume_data
+
+
+def delete_data(k8s_api_client, pod_name):
+    test_data = generate_random_data(0)
+    write_pod_volume_data(k8s_api_client, pod_name, test_data)
+    volume_data = read_volume_data(k8s_api_client, pod_name)
+    assert volume_data == ""
+
+
+def create_snapshot(longhorn_api_client, volume_name):
+    volume = longhorn_api_client.by_id_volume(volume_name)
+
+    snapshots = volume.snapshotList(volume=volume_name)
+    snapshots_count = len(snapshots.data)
+
+    volume.snapshotCreate()
+
+    snapshots = volume.snapshotList(volume=volume_name)
+
+    assert len(snapshots) == snapshots_count + 1
+
+
+def purge_random_snapshot(longhorn_api_client, volume_name, snapshot_name):
+
+    volume = longhorn_api_client.by_id_volume(volume_name)
+
+    volume.snapshotPurge()
+
+    wait_for_snapshot_purge(
+        longhorn_api_client,
+        volume_name,
+        snapshot_name
+    )
+
+
+def delete_random_snapshot(longhorn_api_client, volume_name):
+    volume = longhorn_api_client.by_id_volume(volume_name)
+
+    snapshots = volume.snapshotList(volume=volume_name)
+
+    snapshots_count = len(snapshots.data)
+
+    while True:
+        snapshot_id = randrange(0, snapshots_count)
+
+        if snapshots.data[snapshot_id].name == "volume-head":
+            continue
+        else:
+            break
+    snapshot_name = snapshots.data[snapshot_id].name
+
+    volume.snapshotDelete(name=snapshot_name)
+
+    trigger_purge = randrange(0, 2)
+
+    if trigger_purge == 1:
+        purge_random_snapshot(longhorn_api_client, volume_name, snapshot_name)
+
+
+def test_stress(generate_load):
+    pass
+
+
+@pytest.fixture(params=count)
+def generate_load(request):
+
+    index = request.param
+
+    longhorn_api_client = get_longhorn_api_client()
+    k8s_api_client = get_core_api_client()
+
+    volume_name = STRESS_VOLUME_NAME_PREFIX + index
+    pv_name = STRESS_PV_NAME_PREFIX + index
+    pvc_name = STRESS_PVC_NAME_PREFIX + index
+    pod_name = STRESS_POD_NAME_PREFIX + index
+
+    atexit.register(delete_and_wait_longhorn, longhorn_api_client, volume_name)
+    atexit.register(delete_and_wait_pv, k8s_api_client, pv_name)
+    atexit.register(delete_and_wait_pvc, k8s_api_client, pvc_name)
+    atexit.register(delete_and_wait_pod, k8s_api_client, pod_name)
+
+    create_and_check_volume(
+        longhorn_api_client,
+        volume_name,
+        size=VOLUME_SIZE
+    )
+
+    wait_for_volume_detached(longhorn_api_client, volume_name)
+
+    pv_manifest = generate_pv_maifest_for_longhorn_volume(
+        pv_name,
+        volume_name,
+        VOLUME_SIZE
+    )
+
+    create_and_wait_pv(k8s_api_client, pv_manifest)
```

Comment:
Can we use `create_pv_for_volume()`? The storage class will be `longhorn-static` if you use this function. But I think it doesn't matter since you will manually clean up those components.

The workflow here is similar to `test_kubernetes_status()`, I think you can refer to that test and avoid redundant codes. Same below.

Reply:
fixed